crates/bevy_pbr/src/environment_map/generate_from_skybox.rs

```diff
@@ -0,0 +1,603 @@
+// https://research.activision.com/publications/archives/fast-filtering-of-reflection-probes
+
+use super::{
+    filter_coefficents::{FilterCoefficentsType, FILTER_COEFFICENTS},
+    EnvironmentMapLight, DIFFUSE_CONVOLUTION_SHADER_HANDLE, DOWNSAMPLE_SHADER_HANDLE,
+    FILTER_SHADER_HANDLE,
+};
+use bevy_asset::{Assets, Handle};
+use bevy_core_pipeline::Skybox;
+use bevy_ecs::{
+    prelude::{Component, Entity},
+    query::{QueryItem, Without},
+    system::{Commands, Query, Res, ResMut, Resource},
+    world::{FromWorld, World},
+};
+use bevy_reflect::Reflect;
+use bevy_render::{
+    extract_component::ExtractComponent,
+    render_asset::RenderAssets,
+    render_graph::{NodeRunError, RenderGraphContext, ViewNode},
+    render_resource::{
+        BindGroup, BindGroupDescriptor, BindGroupEntry, BindGroupLayout, BindGroupLayoutDescriptor,
+        BindGroupLayoutEntry, BindingResource, BindingType, Buffer, BufferBindingType,
+        CachedComputePipelineId, ComputePassDescriptor, ComputePipelineDescriptor, Extent3d,
+        FilterMode, PipelineCache, SamplerBindingType, SamplerDescriptor, ShaderStages, ShaderType,
+        StorageTextureAccess, TextureAspect, TextureDescriptor, TextureDimension, TextureFormat,
+        TextureSampleType, TextureUsages, TextureView, TextureViewDescriptor, TextureViewDimension,
+        UniformBuffer,
+    },
+    renderer::{RenderContext, RenderDevice, RenderQueue},
+    texture::{GpuImage, Image, ImageSampler, Volume},
+};
+use bevy_utils::default;
+
+/// Automatically generate an [`EnvironmentMapLight`] from a [`Skybox`].
+///
+/// Usage:
+/// * Add this component via `GenerateEnvironmentMapLight::default()` to an entity with a [`Skybox`] component.
+/// * The first frame this component is added to the skybox entity, an [`EnvironmentMapLight`]
+/// component will be generated and added to the skybox entity.
+/// * For static (non-changing) skyboxes, remove this component 1 frame after adding it to the skybox entity to save performance.
```

Comment:
If the component isn't removed does it regenerate the skybox ibl every frame? It should probably use change detection on the skybox and only run when the skybox changes.

Reply:
Yes. I purposefully did not use change detection as when we get procedurally generated skyboxes, change detection won't work. The texture will be written to each frame by the GPU, but there will be no change detection indicator on the CPU.

Reply:
Is this something we can do for the user, similar to [`PlaybackSettings::Remove`](https://docs.rs/bevy_audio/latest/bevy_audio/enum.PlaybackMode.html)?

Reply:
Could be done, yeah. Not sure how I feel about that idea. Open to hearing other people's opinions.

Reply:
What are the different options when you would expect the light map to be updated? Not expecting it to be included in this PR, just wondering about the use cases.
```rust
enum GenerateEnvironmentMapLightSettings {
    Once,
    EveryFrame,
    Every(Duration),
}
```

Reply:
Probably Once and EveryFrame. I suppose Every(duration) is possible too, but it would probably be more like EveryNFrames instead, duration doesn't make a ton of sense imo.